src/http/headers.cr

```diff
@@ -16,7 +16,14 @@ struct HTTP::Headers
       hasher
     end
 
-    def ==(key2)
+    # Returns true if the normalized name of self is the same as the normalized value passed.
+    #
+    # ```
+    # key1 = HTTP::Headers::Key.new("foo")
+    # key2 = HTTP::Headers::Key.new("FOO")
+    # key1 == key2 #=> true    
+    # ```
+    def ==(key2 : HTTP::Headers::Key) : Bool
```

Comment:
Docs on this method are within a `:nodoc:` type so probably dont need the doc comments since its an implementation detail.

Reply:
Documentation is always great, even if its not for public API :+1:
We probably don't need to bother about examples for such trivial features, though.